docs/DeveloperGuide.md

```diff
@@ -6,33 +6,227 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Overview
+![Overview.png](Overview.png)
```

Comment:
Have you considered the possibility of an incorrect representation of classes and methods in the sequence diagrams?